backends/ubpf/ubpfControl.cpp

```diff
@@ -0,0 +1,772 @@
+/*
+Copyright 2019 Orange
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <p4/enumInstance.h>
+#include "ubpfType.h"
+#include "ubpfControl.h"
+#include "lib/error.h"
+#include "frontends/p4/tableApply.h"
+#include "frontends/p4/typeMap.h"
+#include "frontends/p4/methodInstance.h"
+#include "frontends/p4/parameterSubstitution.h"
+
+namespace UBPF {
+
+    UBPFControlBodyTranslator::UBPFControlBodyTranslator(
+            const UBPFControl *control) :
+            CodeGenInspector(control->program->refMap,
+                             control->program->typeMap), control(control),
+            p4lib(P4::P4CoreLibrary::instance) {
+        setName("UBPFControlBodyTranslator");
+    }
+
+    void UBPFControlBodyTranslator::processFunction(
+            const P4::ExternFunction *function) {
+        if (function->method->name.name == control->program->model.drop.name) {
+            builder->append("pass = false");
```

Comment:
I think you want to use `passVariable` here instead of `pass`.

Reply:
Done.